src/Microsoft.DotNet.Core.Build.Tasks/LockFileCache.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Concurrent;
+using NuGet.Common;
+using NuGet.ProjectModel;
+
+namespace Microsoft.DotNet.Cli.Tasks
+{
+    internal class LockFileCache
```

Comment:
It is persisted statically.  So if the task invocations occur in the same process (or AppDomain in desktop framework), then the cache will be populated.